editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-property-strategies.ts

```diff
@@ -187,6 +186,105 @@ export const convertSizingToVisualSizeWhenPastingFromFlexToFlex =
     ])
   }
 
+export const positionAbsoluteElementComparedToNewParent =
+  (
+    elementToReparent: { oldPath: ElementPath; newPath: ElementPath },
+    targetParent: ElementPath,
+    metadata: ElementInstanceMetadataMap,
+  ): ReparentPropertyStrategy =>
+  () => {
+    if (
+      !MetadataUtils.isPositionAbsolute(
+        MetadataUtils.findElementByElementPath(metadata, elementToReparent.oldPath),
+      )
+    ) {
+      return left('Element is not position: absolute')
+    }
+
+    const targetParentBounds = MetadataUtils.getFrameInCanvasCoords(targetParent, metadata)
+    if (targetParentBounds == null || isInfinityRectangle(targetParentBounds)) {
+      return left('Target parent bounds are invalid')
+    }
+
+    const elementBounds = MetadataUtils.getFrameInCanvasCoords(elementToReparent.oldPath, metadata)
+    if (elementBounds == null || isInfinityRectangle(elementBounds)) {
+      return left('Element bounds are invalid')
+    }
+
+    const deltaX = elementBounds.x - targetParentBounds.x
+    const deltaY = elementBounds.y - targetParentBounds.y
+
+    const elementInBoundsHorizontally = 0 <= deltaX && deltaX <= targetParentBounds.width
+    const elementInBoundsVertically = 0 <= deltaY && deltaY <= targetParentBounds.height
+
+    const horizontalCenter = roundTo((targetParentBounds.width - elementBounds.width) / 2, 0)
+    const verticalCenter = roundTo((targetParentBounds.height - elementBounds.height) / 2, 0)
+
+    const horizontalOffset = elementInBoundsHorizontally ? deltaX : horizontalCenter
+    const verticalOffset = elementInBoundsVertically ? deltaY : verticalCenter
+
+    return right([
+      ...pruneFlexPropsCommands(flexChildProps, elementToReparent.newPath),
+      setCssLengthProperty(
+        'always',
+        elementToReparent.newPath,
+        PP.create('style', 'top'),
+        { type: 'EXPLICIT_CSS_NUMBER', value: cssNumber(verticalOffset, null) },
+        null,
+      ),
+      setCssLengthProperty(
+        'always',
+        elementToReparent.newPath,
+        PP.create('style', 'left'),
+        { type: 'EXPLICIT_CSS_NUMBER', value: cssNumber(horizontalOffset, null) },
+        null,
+      ),
+      setProperty('always', elementToReparent.newPath, styleP('position'), 'absolute'),
+    ])
+  }
+
+const getZIndex = (element: JSXElement): number | null => {
+  const zIndex = getLayoutProperty('zIndex', right(element.props), styleStringInArray)
+  return foldEither(
+    () => null,
+    (z) => z?.value ?? null,
+    zIndex,
+  )
+}
+
+export const setZIndexOnPastedElement =
+  (
+    elementToReparent: { oldPath: ElementPath; newPath: ElementPath },
+    targetParent: ElementPath,
+    metadata: ElementInstanceMetadataMap,
+  ): ReparentPropertyStrategy =>
+  () => {
+    const siblings = MetadataUtils.getChildrenUnordered(metadata, targetParent)
+    const asd = mapDropNulls((sibling) => {
```

Comment:
`asd`?

Reply:
😬 Thanks for noticing it! Must have been left behind from debugging, I removed it in [6eda2ee](https://github.com/concrete-utopia/utopia/pull/3668/commits/6eda2eec9e05e8ee365393ac2448f2ad0e2b78e4)